sdk/storage_datalake/src/file_system.rs

```diff
@@ -41,14 +43,72 @@ pub struct Path {
 }
 
 #[derive(Debug, Clone, Serialize, Deserialize)]
-pub(crate) struct PathList {
+pub struct PathList {
     pub paths: Vec<Path>,
 }
 
 impl TryFrom<Bytes> for PathList {
     type Error = azure_core::error::Error;
 
     fn try_from(response: Bytes) -> Result<Self, Self::Error> {
-        Ok(serde_json::from_slice::<PathList>(response.as_ref())?)
+        Ok(serde_json::from_slice(response.as_ref())?)
+    }
+}
+
+mod i64_or_string {
```

Comment:
In one of our programs we get a quoted value back for bot the `is_directory` and `content_length` fields. In that case serde fails to deserialize. But since encding numbers or bool as strings seems to not be done in other places in the API, i felt it was preferable to just have a custom deserialize that can handle both cases. IN any case we would want to do the conversion from sting... the added code complexity is hopefully bearable... then again there may be things I am overlooking :)